scipy/optimize/_minimize.py

```diff
@@ -108,16 +108,16 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
         obey any specified `bounds`.
     hess : {callable, '2-point', '3-point', 'cs', HessianUpdateStrategy}, optional
         Method for computing the Hessian matrix. Only for Newton-CG, dogleg,
-        trust-ncg,  trust-krylov, trust-exact and trust-constr. If it is
-        callable, it should return the  Hessian matrix:
+        trust-ncg, trust-krylov, trust-exact and trust-constr. If it is
+        callable, it should return the Hessian matrix:
 
             ``hess(x, *args) -> {LinearOperator, spmatrix, array}, (n, n)``
 
         where x is a (n,) ndarray and `args` is a tuple with the fixed
-        parameters. LinearOperator and sparse matrix returns are
-        allowed only for 'trust-constr' method. Alternatively, the keywords
+        parameters. LinearOperator and sparse matrix returns are only allowed
```

Comment:
Maybe reference the object here?
```suggestion
        parameters. `scipy.sparse.linalg.LinearOperator` and sparse matrix returns are only allowed
```

Reply:
I'd prefer as is. It looks a bit silly to write out the entire path. Is there a way to get it to display as a class without writing out the entire path?

Reply:
> I'd prefer as is. It looks a bit silly to write out the entire path. Is there a way to get it to display as a class without writing out the entire path?

Looking at the extensions listed in [scipy's configuration file for sphinx ](https://github.com/scipy/scipy/blob/master/doc/source/conf.py) I believe there are two options. 

If you would like it to be displayed as `LinearOperator` then the qualifier `~` can be included to truncate everything except the base name (it would still appear in the raw docs, clearly).
```
:class:`~scipy.sparse.linalg.LinearOperator`
```
should appear as `LinearOperator`

The only other alternative I know is to use a directive to define the current module
```
.. currentmodule:: scipy.sparse.linalg
```
so that `LinearOperator` could be referenced directly, but this should really only be used to document the index file being used to describe the `scipy.sparse.linalg` module itself.

Reply:
Indeed, nice. But again, optional.